examples/schelling_segregation/src/main.cu

```diff
@@ -0,0 +1,357 @@
+    for (const auto& message : FLAMEGPU->message_in(FLAMEGPU->getID() - 1)) {
+        flamegpu::id_t winning_id = message.getVariable<flamegpu::id_t>("id");
+        FLAMEGPU->setVariable<unsigned int>("next_type", message.getVariable<unsigned int>("type"));
+        FLAMEGPU->setVariable<unsigned int>("available", 0);
+        FLAMEGPU->message_out.setIndex(winning_id - 1);
+        FLAMEGPU->message_out.setVariable<unsigned int>("won", 1);
+        break;
+    }
+    return flamegpu::ALIVE;
+}
+
+FLAMEGPU_AGENT_FUNCTION(has_moved, flamegpu::MessageArray, flamegpu::MessageNone) {
+    const auto& message = FLAMEGPU->message_in.at(FLAMEGPU->getID() - 1);
+    if (message.getVariable<unsigned int>("won")) {
+        FLAMEGPU->setVariable<unsigned int>("movement_resolved", 1);
+    }
+    return flamegpu::ALIVE;
+}
+
+FLAMEGPU_EXIT_CONDITION(movement_resolved) {
+    return (FLAMEGPU->agent("agent").count<unsigned int>("movement_resolved", 0) == 0) ? flamegpu::EXIT : flamegpu::CONTINUE;
+}
+
+FLAMEGPU_AGENT_FUNCTION(update_locations, flamegpu::MessageNone, flamegpu::MessageNone) {
+    FLAMEGPU->setVariable<unsigned int>("type", FLAMEGPU->getVariable<unsigned int>("next_type"));
+    return flamegpu::ALIVE;
+}
+
+int main(int argc, const char ** argv) {
+    NVTX_RANGE("main");
+    NVTX_PUSH("ModelDescription");
+
+    flamegpu::ModelDescription model("Schelling_segregation");
+
+    /**
+     * Messages
+     */
+    {
+        {
+            flamegpu::MessageArray2D::Description &message = model.newMessage<flamegpu::MessageArray2D>("type_message");
+            message.newVariable<unsigned int>("type");
+            message.setDimensions(GRID_WIDTH, GRID_WIDTH);
+        }
+    }
+
+    /**
+     * Agents
+     */
+    {   // Per cell agent
+        flamegpu::AgentDescription  &agent = model.newAgent("agent");
+        agent.newVariable<unsigned int, 2>("pos");
+        agent.newVariable<unsigned int>("type");
+        agent.newVariable<unsigned int>("next_type");
+        agent.newVariable<unsigned int>("happy");
+        agent.newVariable<unsigned int>("available");
+        agent.newVariable<unsigned int>("movement_resolved");
+#ifdef VISUALISATION
+        // Redundant seperate floating point position vars for vis
+        agent.newVariable<float>("x");
+        agent.newVariable<float>("y");
+#endif
+        // Functions
+        agent.newFunction("output_type", output_type).setMessageOutput("type_message");
+        agent.newFunction("determine_status", determine_status).setMessageInput("type_message");
+        agent.newFunction("update_locations", update_locations);
+    }
+
+    /**
+     * Movement resolution submodel
+     */
+
+    flamegpu::ModelDescription submodel("plan_movement");
+    submodel.addExitCondition(movement_resolved);
+    {
+        // Environment
+        {
+            flamegpu::EnvironmentDescription& env = submodel.Environment();
+            env.newProperty<unsigned int>("spaces_available", 0);
```

Comment:
If this is intended to be user configurable/accessible, you will need to define it in the parent model and map it too.

Reply:
*On reviewing, I don't think it is, so this can be marked resolved.*